frontend/src/metabase/core/components/TabList/TabList.tsx

```diff
@@ -23,20 +27,72 @@ const TabList = forwardRef(function TabGroup<T>(
 ) {
   const idPrefix = useUniqueId();
   const outerContext = useContext(TabContext);
+
+  const [scrollPosition, setScrollPosition] = useState(0);
+  const [showScrollLeft, setShowScrollLeft] = useState(false);
+  const [showScrollRight, setShowScrollRight] = useState(false);
+
+  const tabListContentRef = useRef(null);
+
   const innerContext = useMemo(() => {
     return { value, idPrefix, onChange };
   }, [value, idPrefix, onChange]);
+
   const activeContext = outerContext.isDefault ? innerContext : outerContext;
 
+  const scroll = (direction: string) => {
+    if (tabListContentRef.current) {
+      const container = tabListContentRef.current as HTMLDivElement;
+
+      const scrollDistance =
+        (container.offsetWidth - 32) * (direction === "left" ? -1 : 1);
+      container.scroll(container.scrollLeft + scrollDistance, 0);
+      setScrollPosition(container.scrollLeft + scrollDistance);
+    }
+  };
+
+  useEffect(() => {
+    if (!tabListContentRef.current) {
+      return;
+    }
+
+    const container = tabListContentRef.current as HTMLDivElement;
+    setShowScrollRight(
+      scrollPosition + container.offsetWidth < container.scrollWidth,
+    );
+    setShowScrollLeft(scrollPosition > 0);
+  }, [setShowScrollLeft, setShowScrollRight, scrollPosition]);
+
   return (
     <TabListRoot {...props} ref={ref} role="tablist">
-      <TabListContent>
+      <TabListContent ref={tabListContentRef}>
         <TabContext.Provider value={activeContext as TabContextType}>
           {children}
         </TabContext.Provider>
       </TabListContent>
+      {showScrollLeft && (
+        <ScrollArrow direction="left" onClick={() => scroll("left")} />
```

Comment:
Not sure if we care with non-DOM props being passed at the moment.
![image](https://user-images.githubusercontent.com/1937582/177300581-1f9f8136-4cd3-4552-8894-1a3d77f414a3.png)

I found @losrebellos already made a solution for this.

https://github.com/metabase/metabase/blob/8b7022a5f76695717da58d5dbb43b041aa97b0a4/frontend/src/metabase/components/Badge.styled.jsx#L40-L42

@ranquild Is it fine passing non-DOM props to DOM elements? I feel like ideally not, but it would require quite a boilerplate as emotion doesn't implement transient props [like styled-component does](https://styled-components.com/docs/api#transient-props).

Reply:
@WiNloSt emotion does it a bit differently https://emotion.sh/docs/styled#customizing-prop-forwarding. I think it should be enough to just rename to prop to have 2 words in camel case, e.g. `scrollDirection`, for it to be automatically excluded.

Reply:
Hah, maybe this is just a one-off case. Emotion declares `direction` as a valid DOM attribute. https://github.com/emotion-js/emotion/blob/7bf1d0e007250009774159abe7550e3167bcd533/packages/is-prop-valid/src/props.js#L283

But I couldn't find any source that confirm `direction` as a valid attribute e.g. [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes)